bundles/org.openhab.core.model.rule.runtime/src/org/openhab/core/model/rule/runtime/internal/DSLRuleProvider.java

```diff
@@ -264,14 +265,26 @@ private Rule toRule(String modelName, boolean hasContext, org.openhab.core.model
         XBlockExpression expression = rule.getScript();
         String script = NodeModelUtils.findActualNodeFor(expression).getText();
         Configuration cfg = new Configuration();
-        cfg.put("script", context + script);
+        cfg.put("script", context + removeIndentation(script));
         cfg.put("type", MIMETYPE_OPENHAB_DSL_RULE);
         List<Action> actions = Collections.singletonList(ActionBuilder.create().withId("script")
                 .withTypeUID("script.ScriptAction").withConfiguration(cfg).build());
 
         return RuleBuilder.create(uid).withName(name).withTriggers(triggers).withActions(actions).build();
     }
 
+    private String removeIndentation(String script) {
+        String s = script;
+        if (s.startsWith("\n")) {
+            s = s.substring(1);
+        }
```

Comment:
This doesn't work on Windows, you'll want to add this as well:
```java
        if (s.startsWith("\r\n")) {
            s = s.substring(2);
        }
```
Maybe also add a final newline if it doesn't exist (that would make the YAML multiline control a simple `|` instead of `|-`)

Reply:
These Windows users ;-)
Done!